DependencyInjection/DoctrineExtension.php

```diff
@@ -107,6 +109,40 @@ public function load(array $configs, ContainerBuilder $container)
         $this->ormLoad($config['orm'], $container);
     }
 
+    /**
+     * Process user configuration and adds a default DBAL connection and/or a
+     * default EM if required, then process again the configuration to get
+     * default values for each.
+     *
+     * @param array<array<mixed>> $configs
+     *
+     * @return array<mixed>
+     */
+    private function processConfigurationPrependingDefaults(ConfigurationInterface $configuration, array $configs): array
+    {
+        $config      = $this->processConfiguration($configuration, $configs);
+        $configToAdd = [];
```

Comment:
Why this instead of doing `$configs[] =` ?

Reply:
if we want to call `processConfiguration` a second time only when required we need to track if some configuration has been added or not.
if we directly append to `configs` we can't really tell 🤷

Reply:
But thinking about it, we can use a Boolean value to track this if you prefer